block/cardano-sl-block.cabal

```diff
@@ -132,3 +128,82 @@ library
 
   build-tools: cpphs >= 1.19
   ghc-options: -pgmP cpphs -optP --cpp
+
+test-suite test
+  main-is:             test.hs
+  hs-source-dirs:      test
+  default-language:    Haskell2010
+  type:                exitcode-stdio-1.0
+
+  other-modules:
+                       Spec
+                       Test.Pos.Block.Arbitrary
+                       Test.Pos.Block.Arbitrary.Message
+                       Test.Pos.Block.CborSpec
+
+  build-depends:       base
+                     , bytestring
+                     , cardano-sl-binary
+                     , cardano-sl-binary-test
+                     , cardano-sl-block
+                     , cardano-sl-core
+                     , cardano-sl-core-test
+                     , cardano-sl-crypto
+                     , cardano-sl-crypto-test
+                     , cardano-sl-delegation-test
+                     , cardano-sl-ssc
+                     , cardano-sl-txp-test
+                     , cardano-sl-update
+                     , cardano-sl-update-test
+                     , cardano-sl-util
+                     , cardano-sl-util-test
+                     , containers
+                     , formatting
+                     , generic-arbitrary
+                     , hspec
+                     , QuickCheck
+                     , quickcheck-instances
+                     , random
+                     , serokell-util >= 0.1.3.4
+                     , text
+                     , text-format
+                     , time-units
+                     , universum >= 0.1.11
+                     , unordered-containers
+
+  ghc-options:         -threaded
+                       -rtsopts
+                       -Wall
+
+  -- linker speed up for linux
+  if os(linux)
+    ghc-options:       -optl-fuse-ld=gold
+    ld-options:        -fuse-ld=gold
+
+  default-extensions:   DeriveDataTypeable
+                        DeriveGeneric
+                        GeneralizedNewtypeDeriving
+                        StandaloneDeriving
+                        FlexibleContexts
+                        FlexibleInstances
+                        MultiParamTypeClasses
+                        FunctionalDependencies
+                        DefaultSignatures
+                        NoImplicitPrelude
+                        OverloadedStrings
+                        RecordWildCards
+                        TypeApplications
+                        TupleSections
+                        ViewPatterns
+                        LambdaCase
+                        MultiWayIf
+                        ConstraintKinds
+                        UndecidableInstances
+                        BangPatterns
+                        TemplateHaskell
+                        ScopedTypeVariables
+                        GADTs
+                        MonadFailDesugaring
```

Comment:
This could be trimmed down, if we are still trying to move slowly in that direction.